### PR TITLE
INSTR() func

### LINKED ISSUE #3044: INSTR() | Return the index of the first occurrence of substring

### PR DESCRIPTION
**Pull request by mshauneu**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Impl of INSTR() func

## Changelog

- New Feature

## Related Issues

Fixes #3044 

## Test Plan

Stateless Tests

### CODE REVIEW

File: common/functions/src/scalars/strings/locate.rs [hunk 1]
```diff
@@ -62,8 +62,18 @@ impl Function for LocateFunction {
     }
 
     fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
-        let ss_column = columns[0].column().cast_with_type(&DataType::String)?;
-        let s_column = columns[1].column().cast_with_type(&DataType::String)?;
+        let (ss_column, s_column) = if *self.display_name.to_uppercase() == "INSTR".to_owned() {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/locate.rs:65]: A generic constant value will be better.

Comment (mshauneu) [reply common/functions/src/scalars/strings/locate.rs:65]: @sundy-li can you explain more?

Comment (mshauneu) [reply common/functions/src/scalars/strings/locate.rs]: you are right. fixed.

File: common/functions/src/scalars/strings/locate.rs [hunk 2]
```diff
@@ -22,14 +22,54 @@ use crate::scalars::function_factory::FunctionDescription;
...
+    }
+
+    pub fn desc() -> FunctionDescription {
+        LocatingFunction::<FUNC_LOCATE>::desc()
+    }
+}
+
+pub struct InstrFunction {}
+
+impl InstrFunction {
```
Comment (sundy-li) [inline common/functions/src/scalars/strings/locate.rs:43]: type LocateFunction =  LocatingFunction::<FUNC_LOCATE>;
type InstrFunction  = LocatingFunction::<FUNC_INSTR>;

Comment (mshauneu) [reply common/functions/src/scalars/strings/locate.rs]: thank you, changed

**Review by sundy-li — APPROVED** [review]

File: common/functions/src/scalars/strings/locate.rs (hunk 2 above)
Comment (sundy-li) [reply common/functions/src/scalars/strings/locate.rs]: Besides, `String2String` function is a good template function to learn.

**Pull request merged**